lib/src/client.rs

```diff
@@ -241,17 +240,14 @@ impl ClientInner {
 pub struct Client {
     inner: Arc<ClientInner>,
     consensus: Option<Consensus>,
+    #[cfg(feature = "validator")]
     validator: Option<Validator>,
 }
 
 impl Client {
     pub async fn from_config(config: ClientConfig) -> Result<Self, Error> {
-        let (inner, consensus, validator) = ClientInner::from_config(config).await?;
-        Ok(Client {
-            inner: Arc::new(inner),
-            consensus: Some(consensus),
-            validator,
-        })
+        let client: Result<Self, Error> = ClientInner::from_config(config).await;
+        client
```

Comment:
This could be written more idiomatically as `ClientInner::from_config(config).await` and it works on my environment, did you had issues doing it that way?

Reply:
I thought so but now that I changed it, it worked